app/livechat/server/api/lib/visitors.js

```diff
@@ -41,3 +41,34 @@ export async function findVisitedPages({ userId, roomId, pagination: { offset, c
 		total,
 	};
 }
+
+export async function findChatHistory({ userId, roomId, visitorId, pagination: { offset, count, sort } }) {
+	if (!await hasPermissionAsync(userId, 'view-l-room')) {
+		throw new Error('error-not-authorized');
+	}
+	const room = await LivechatRooms.findOneById(roomId);
+	if (!room) {
+		throw new Error('invalid-room');
+	}
+	const subscription = Subscriptions.findOneByRoomIdAndUserId(room._id, this.userId, { fields: { _id: 1 } });
+	if (!subscription) {
+		throw new Error('invalid-room');
+	}
+
+	const cursor = await LivechatRooms.findByVisitorId(visitorId, {
```

Comment:
What's the reason for this `await` since the method doesn't return a `promise`?
The same question for the other `await's` below.